src/Sentry/Sentry.csproj

```diff
@@ -45,6 +45,32 @@
     </Compile>
   </ItemGroup>
 
+  <!-- Profiling is supported only for select TargetFrameworks -->
+  <PropertyGroup Condition="'$(TargetFramework)' == 'net6.0'">
+    <_ProfilingSupported>true</_ProfilingSupported>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>    <!-- Required for TraceEvent -->
+  </PropertyGroup>
+
+  <!-- Profiling dependencies are compiled directly into Sentry. -->
+  <ItemGroup Condition="'$(_ProfilingSupported)' == 'true'">
+    <Compile Include="../../modules/perfview/src/TraceEvent/**/*.cs">
+      <Link>%(RecursiveDir)\%(Filename)%(Extension)</Link>
+    </Compile>
+    <Compile Include="../../modules/perfview/src/FastSerialization/**/*.cs">
+      <Link>%(RecursiveDir)\%(Filename)%(Extension)</Link>
+    </Compile>
+    <Compile Include="../../modules/perfview/src/Utilities/**/*.cs">
+      <Link>%(RecursiveDir)\%(Filename)%(Extension)</Link>
+    </Compile>
+    <!-- Exclude samples, tests and some problematic files -->
+    <Compile Remove="../../modules/perfview/src/TraceEvent/Samples/**/*" />
+    <Compile Remove="../../modules/perfview/src/TraceEvent/Properties/**/*" />
+    <Compile Remove="../../modules/perfview/src/TraceEvent/AutomatedAnalysis/**/*" />
+    <Compile Remove="../../modules/perfview/src/**/*.Tests/**/*" />
+    <Compile Remove="../../modules/perfview/src/TraceEvent/ETWReloggerTraceEventSource.cs" />
+    <Compile Remove="../../modules/perfview/src/TraceEvent/Symbols/NativeSymbolModule.cs" />
```

Comment:
How did you come up with those in particular?

Reply:
And to answer your question - just removing stuff we don't need in our project. How exactly: try to compile, look at errors, update tools, repeat...